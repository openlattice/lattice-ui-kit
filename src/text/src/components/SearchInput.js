/*
 * @flow
 */

import React from 'react';

import styled from 'styled-components';
import { faSearch } from '@fortawesome/pro-regular-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import inputStyles from './styled/inputStyles';
import { NEUTRALS } from '../../../colors';

const SearchInputWrapper = styled.div`
  display: flex;
  position: relative;
`;

const Input = styled.input`
  ${inputStyles}
  color: ${NEUTRALS[0]};
  padding-left: 32px;

  &::placeholder {
    color: ${NEUTRALS[1]};
  }
`;

const SearchIcon = styled.div`
  align-self: center;
  color: ${NEUTRALS[1]};
  margin-left: 11px;
  font-size: 14px;
  line-height: 14px;
  position: absolute;
`;

type Props = {
  disabled ?:boolean;
  id ?:string;
  name ?:string;
  onBlur ?:(event :SyntheticFocusEvent<HTMLInputElement>) => void;
  onChange ?:(event :SyntheticInputEvent<HTMLInputElement>) => void;
  onFocus ?:(event :SyntheticFocusEvent<HTMLInputElement>) => void;
  placeholder ?:string;
  readOnly ?:boolean;
  value ?:any;
};

const SearchInput = (props :Props) => (
  <SearchInputWrapper>
    <SearchIcon>
      <FontAwesomeIcon icon={faSearch} />
    </SearchIcon>
    <Input {...props} type="search" />
  </SearchInputWrapper>
);

SearchInput.defaultProps = {
  disabled: false,
  id: undefined,
  name: undefined,
  onBlur: undefined,
  onChange: undefined,
  onFocus: undefined,
  placeholder: 'Search...',
  readOnly: undefined,
  value: undefined,
};

export default SearchInput;
