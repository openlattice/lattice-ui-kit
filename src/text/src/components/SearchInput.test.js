import React from 'react';

import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import SearchInput from './SearchInput';

describe('SearchInput', () => {

  describe('snapshots', () => {

    test('should match snapshot', () => {
      const wrapper = mount(<SearchInput />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

  });

  describe('props', () => {

    test('should default to type="search"', () => {
      const wrapper = mount(<SearchInput />);
      expect(wrapper.find('input').prop('type')).toEqual('search');
    });

    test('should ignore type prop and default to type="search"', () => {
      const wrapper = mount(<SearchInput type="password" />);
      expect(wrapper.find('input').prop('type')).toEqual('search');
    });

    test('should pass props through', () => {
      const props = {
        autoComplete: 'on',
        disabled: true,
        form: 'test-form',
        max: 8,
        maxLength: 8,
        min: 1,
        name: 'test',
        pattern: '/.+/',
        placeholder: 'test placeholder',
        required: true,
      };
      const wrapper = mount(<SearchInput {...props} />);
      expect(wrapper.find('input').props()).toEqual(expect.objectContaining(props));
    });

  });

});
