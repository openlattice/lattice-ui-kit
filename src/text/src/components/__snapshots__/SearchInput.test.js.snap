// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput snapshots should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c2 {
  background-color: #f9f9fd;
  border-radius: 3px;
  border: 1px solid #dcdce7;
  box-shadow: none;
  box-sizing: border-box;
  color: #555e6f;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  font-size: 14px;
  min-height: 40px;
  padding: 10px;
  text-overflow: ellipsis;
  -webkit-transition: background-color 200ms ease-out,border-color 200ms ease-out;
  transition: background-color 200ms ease-out,border-color 200ms ease-out;
  width: 100%;
  color: #555e6f;
  padding-left: 32px;
}

.c2:hover {
  background-color: #f0f0f7;
}

.c2:focus {
  border: solid 1px #6124e2;
  background-color: white;
  outline: none;
}

.c2:disabled {
  background-color: #f9f9fd;
  color: #8e929b;
  cursor: not-allowed;
}

.c2::-webkit-input-placeholder {
  color: #8e929b;
}

.c2::-moz-placeholder {
  color: #8e929b;
}

.c2:-ms-input-placeholder {
  color: #8e929b;
}

.c2::placeholder {
  color: #8e929b;
}

.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #8e929b;
  margin-left: 11px;
  font-size: 14px;
  line-height: 14px;
  position: absolute;
}

<SearchInput
  disabled={false}
  placeholder="Search..."
>
  <SearchInput__SearchInputWrapper>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "SearchInput__SearchInputWrapper-sc-1bpx0dr-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "display:flex;position:relative;",
            ],
          },
          "displayName": "SearchInput__SearchInputWrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "SearchInput__SearchInputWrapper-sc-1bpx0dr-0",
          "target": "div",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <SearchInput__SearchIcon>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SearchInput__SearchIcon-sc-1bpx0dr-2",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "align-self:center;color:",
                    "#8e929b",
                    ";margin-left:11px;font-size:14px;line-height:14px;position:absolute;",
                  ],
                },
                "displayName": "SearchInput__SearchIcon",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "SearchInput__SearchIcon-sc-1bpx0dr-2",
                "target": "div",
                "toString": [Function],
                "usesTheme": false,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c1"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f002",
                      "M508.5 468.9L387.1 347.5c-2.3-2.3-5.3-3.5-8.5-3.5h-13.2c31.5-36.5 50.6-84 50.6-136C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c52 0 99.5-19.1 136-50.6v13.2c0 3.2 1.3 6.2 3.5 8.5l121.4 121.4c4.7 4.7 12.3 4.7 17 0l22.6-22.6c4.7-4.7 4.7-12.3 0-17zM208 368c-88.4 0-160-71.6-160-160S119.6 48 208 48s160 71.6 160 160-71.6 160-160 160z",
                    ],
                    "iconName": "search",
                    "prefix": "far",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-search fa-w-16 "
                  data-icon="search"
                  data-prefix="far"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M508.5 468.9L387.1 347.5c-2.3-2.3-5.3-3.5-8.5-3.5h-13.2c31.5-36.5 50.6-84 50.6-136C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c52 0 99.5-19.1 136-50.6v13.2c0 3.2 1.3 6.2 3.5 8.5l121.4 121.4c4.7 4.7 12.3 4.7 17 0l22.6-22.6c4.7-4.7 4.7-12.3 0-17zM208 368c-88.4 0-160-71.6-160-160S119.6 48 208 48s160 71.6 160 160-71.6 160-160 160z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
            </div>
          </StyledComponent>
        </SearchInput__SearchIcon>
        <SearchInput__Input
          disabled={false}
          placeholder="Search..."
          type="search"
        >
          <StyledComponent
            disabled={false}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SearchInput__Input-sc-1bpx0dr-1",
                  "isStatic": false,
                  "lastClassName": "c2",
                  "rules": Array [
                    "background-color:",
                    "#f9f9fd",
                    ";border-radius:3px;border:1px solid ",
                    [Function],
                    ";box-shadow:none;box-sizing:border-box;color:",
                    "#555e6f",
                    ";display:flex;flex:0 1 auto;font-size:14px;min-height:40px;padding:10px;text-overflow:ellipsis;transition:background-color ",
                    "200ms",
                    " ease-out,border-color ",
                    "200ms",
                    " ease-out;width:",
                    [Function],
                    ";:hover{background-color:",
                    "#f0f0f7",
                    ";}:focus{border:solid 1px ",
                    "#6124e2",
                    ";background-color:white;outline:none;}:disabled{background-color:",
                    "#f9f9fd",
                    ";color:",
                    "#8e929b",
                    ";cursor:not-allowed;}",
                    " color:",
                    "#555e6f",
                    ";padding-left:32px;&::placeholder{color:",
                    "#8e929b",
                    ";}",
                  ],
                },
                "displayName": "SearchInput__Input",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "SearchInput__Input-sc-1bpx0dr-1",
                "target": "input",
                "toString": [Function],
                "usesTheme": false,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            placeholder="Search..."
            type="search"
          >
            <input
              className="c2"
              disabled={false}
              placeholder="Search..."
              type="search"
            />
          </StyledComponent>
        </SearchInput__Input>
      </div>
    </StyledComponent>
  </SearchInput__SearchInputWrapper>
</SearchInput>
`;
